src/Emulator/Peripherals/Peripherals/DMA/ATCDMAC100.cs

```diff
@@ -0,0 +1,578 @@
+                        case BlockSizeMode.Unit16:
+                            return 1u << (byte)descriptor.blockSize;
+                        case BlockSizeMode.Unit32:
+                            return 1u << (byte)descriptor.blockSize;
+                        case BlockSizeMode.Unit64:
+                            return 1u << (byte)descriptor.blockSize;
+                        case BlockSizeMode.Unit128:
+                            return 1u << (byte)descriptor.blockSize;
+                        default:
+                            parent.Log(LogLevel.Warning, "Channel #{0} Invalid Block Size Mode value.", Index);
+                            return 0;
+                    }
+                }
+            }
+
+            private uint TranSize => (uint)descriptor.TranSize;
+            private ulong LinkStructureAddress => (ulong)descriptor.linkAddress << 2;
+            private uint SourceIncrement => descriptor.SrcAddrCtrl == AddressMode.Fixed ? 0u : ((1u << (byte)descriptor.srcwidth));
+            private uint DestinationIncrement => descriptor.DstAddrCtrl == AddressMode.Fixed ? 0u : ((1u << (byte)descriptor.dstwidth));
+            private TransferType SizeAsTransferType => (TransferType)(1 << (byte)descriptor.srcwidth);
+            private int Bytes => (int)Math.Min(TranSize, BlockSizeMultiplier) << (byte)descriptor.srcwidth;
+            private bool SourceIncrementMode => descriptor.SrcAddrCtrl == AddressMode.Increment;
+            private bool SourceDecrementMode => descriptor.SrcAddrCtrl == AddressMode.Decrement;
+            private bool DestinationIncrementMode => descriptor.DstAddrCtrl == AddressMode.Increment;
+            private bool DestinationDecrementMode => descriptor.DstAddrCtrl == AddressMode.Decrement; 
+
+            private Descriptor descriptor;
+            private ulong? descriptorAddress;
+            private bool isInProgress;
+            private IFlagRegisterField InterruptTCMask;
+            private readonly ATCDMAC100 parent;
+            protected readonly int DescriptorSize = Packet.CalculateLength<Descriptor>();
+
+            protected enum BlockSizeMode
+            {
+                Unit1 = 0x0,
+                Unit2 = 0x1,
+                Unit4 = 0x2,
+                Unit8 = 0x3,
+                Unit16 = 0x4,
+                Unit32 = 0x5,
+                Unit64 = 0x6,
+                Unit128 = 0x7,
+
+            }
+
+            protected enum AddressMode
+            {
+                Increment = 0x0,
+                Decrement = 0x1,
+                Fixed = 0x2,
+                Reserved = 0x3,
+            }
+
+            protected enum SizeMode
+            {
+                Byte = 0x0,
+                HalfWord = 0x1,
+                Word = 0x2,
+
+            }
+
+            [LeastSignificantByteFirst]
+            private struct Descriptor
+            {
+                public string PrettyString => $@"Descriptor {{        
+    Enable: {Enabled},
```

Comment:
This part of the code should be better indented. Funcionally speaking this doesn't matter but remember that to code is some sort of art, so it should look beautiful

Reply:
Adopt this code style from EFR32MG12_LDMA as that is used as a reference point.